framework/src/ics/InitialConditionWarehouse.C

```diff
@@ -33,40 +33,45 @@ InitialConditionWarehouse::initialSetup(THREAD_ID tid)
 void
 InitialConditionWarehouse::addObject(MooseSharedPointer<InitialCondition> object, THREAD_ID tid)
 {
-  // Check that when object is boundary restricted that the variable is nodal
-  const MooseVariable & var = object->variable();
 
-  // Boundary Restricted
-  if (object->boundaryRestricted())
+  // Perform IC error checking, there is no need to do this on each thread, since they are all the same
```

Comment:
While this is true we don't don't restrict error checking anywhere else in the code even when they all threads have the same information. If this is truly failing in threaded mode _only_ then we should figure out why rather than simply skipping this check. If for some reason the threads don't have the same information then this will lead to a seg. fault or another much more difficult to find bug later.